src/packages/recompose/componentFromProp.js

```diff
@@ -1,9 +1,8 @@
 import { createElement } from 'react'
-import omit from './utils/omit'
 
 const componentFromProp = propName => {
-  const Component = props =>
-    createElement(props[propName], omit(props, [propName]))
+  const Component = ({ [propName]: elementType, forwardedRef, ...props }) =>
```

Comment:
```suggestion
  const Component = ({ [propName], forwardedRef, ...props }) =>
```

I would generally not favor including the type info in the JavaScript. I think it would be ideal to look into using JSDoc comments to include the type info and enable type checking with TypeScript (there are quite a few libraries doing this now, take a look at Prettier for example).

And can we better explain what happens when:

- there is no `forwardedRef` prop here?
- someone uses `ref` instead of `forwardedRef`, before and after this change?
- someone uses both `ref` & `forwardedRef`?

Reply:
Thanks for the input! I suppose I'm a little unsure what you mean about including type info. 🤔 There isn't any "type info," per se. I just used an alias for the first param to `React.createElement()` (which the React docs call simply `type`), in order to make the existing behavior a bit more readable while simultaneously eliminating the need to `omit()` the propName from the forwarded props.

Reply:
> There isn't any "type info," per se. I just used an alias [...]

Yeah my bad, guess I've been using ES3 & ES5 for way too long!

Reply:
I think it would still be good if we can answer the questions from my earlier comment:

> - there is no `forwardedRef` prop here?
> - someone uses `ref` instead of `forwardedRef`, before and after this change?
> - someone uses both `ref` & `forwardedRef`?